src/union.rs

```diff
@@ -62,7 +62,7 @@ pub fn digraph_union(
         node_map.insert(node.index(), node_index);
     }
 
-    for edge in b.weighted_edge_list(py) {
+    for edge in b.weighted_edge_list(py).edges {
```

Comment:
Why is this the only place where `.edges` is needed? Is this the only place where `weighted_edge_list()` was used?

Reply:
Yeah, this is the only place `weighted_edge_list()` is called from inside rust. Normally it's only used from python but this function uses it instead of calling `b.graph..edge_references()`. We probably should update this to just use the petgraph api directly, but it's not a big deal and an easy followup thing we can change later.